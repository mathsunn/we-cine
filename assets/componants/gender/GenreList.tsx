import React from "react";
import useGenres from "../../hooks/UseGenre";
import { map } from "lodash";
import { Genre } from "../../contracts/Genre";
import { useFilters } from "../../context/FilterContext";

const GenreList: React.FC = () => {
  const { genres, total, loading } = useGenres();
  const { selectedGenre, setGenre } = useFilters();

  const handleClick = (genre: Genre) => {
    setGenre(genre);
  };

  if (loading) {
    return <>Loading</>;
  }

  if (total === 0) {
    return <>No result</>;
  }

  return (
    <>
      {map(genres, (genre) => (
        <div key={genre.id}>
          <button
            id={`genre_${genre.id}`}
            className="btn btn-link"
            disabled={selectedGenre && genre.id === selectedGenre.id}
            onClick={() => handleClick(genre)}
          >
            {genre.name}
          </button>
        </div>
      ))}
    </>
  );
};

export default GenreList;
