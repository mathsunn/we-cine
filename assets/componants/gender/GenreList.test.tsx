import React from "react";
import { render } from "@testing-library/react";
import GenreList from "./GenreList";
import axios from "axios";

jest.mock("axios");
const mockedAxios = axios as jest.Mocked<typeof axios>;

test("matches the snapshot", async () => {
  mockedAxios.get.mockResolvedValue({ data: [{ id: 28, name: "Action" }] });

  const { asFragment, findByText } = render(<GenreList />);

  await findByText(/Action/i);

  expect(asFragment()).toMatchSnapshot();
});
