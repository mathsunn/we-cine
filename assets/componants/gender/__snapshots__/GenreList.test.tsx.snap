// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn btn-link"
      id="genre_28"
    >
      Action
    </button>
  </div>
</DocumentFragment>
`;
